### DISCUSSION TITLE
Accessing the contents of a bulk RPC

### DISCUSSION THREAD

**Discussion by rkho** [supabase/supabase#2354] [Questions] [answered]
I've a SQL function to increment a given row matching its ID:

```
create or replace function increment_select_values(row_id uuid)
returns void as
$$
  update select_values_map
  set num_chosen = num_chosen + 1
    where select_uuid = row_id
$$ 
language sql volatile;
```

And I'm trying out the Supabase `rpc` method:

```
      const { data, error } = await supabase
        .rpc('increment_select_values', [
          { row_id: '00a8848e-9118-455b-93ce-cd2515f64401' },
          { row_id: '13759a79-7dee-4a88-8621-6c8c92d3ffb4' },
        ]);
```

However, only the **first** row_id has its count modified.

How should I update the function above to account for a list of row_ids?

**Comment by dragarcia** [comment, marked as answer]
Hi @rkho 👋 

You could try this:
```sql
create or replace function increment_select_values(row_ids uuid[])
returns void as
$$
  update select_values_map
  set num_chosen = num_chosen + 1
    where select_uuid in unnest(row_ids)
$$ 
language sql volatile;
```

- The function now expects it to be supplied with an array of `uuid`
- the function [unnest()](https://www.postgresql.org/docs/current/functions-array.html) converts an array into rows.

Then, with the `rpc` method:
```
const { data, error } = await supabase
    .rpc('increment_select_values', [
        '00a8848e-9118-455b-93ce-cd2515f64401',
        '13759a79-7dee-4a88-8621-6c8c92d3ffb4'
    ]);
```

**Reply by rkho** [reply]
Thanks for the followup; this proposed solution is giving me syntax problems:

```
create or replace function increment_select_values(row_ids uuid[])
returns void as
$$
  update select_values_map
  set num_chosen = num_chosen + 1
    where select_uuid in unnest(row_ids)
$$ 
language sql volatile;
```

> syntax error at or near "unnest"

I tried to remove the unnest call and also tried to iterate through the row_ids to no avail

**Reply by steve-chavez** [reply]
@rkho  Change the `where select_uuid in unnest(row_ids)` to `where select_uuid = any(row_ids)`.

**Reply by dragarcia** [reply]
Ah yes, my bad. Thanks @steve-chavez! Here's the overall one with Steve's change:

```sql
create or replace function increment_select_values(row_ids uuid[])
returns void as
$$
  update select_values_map
  set num_chosen = num_chosen + 1
    where select_uuid = any(row_ids)
$$ 
language sql volatile;
```
